build.sbt

```diff
@@ -559,12 +553,39 @@ lazy val publishableSettings = Def.settings(
     false
   },
   mimaPreviousArtifacts := {
-    // Compare against 3.0.0 throughtout the 3.x series.
-    val stableVersion = "3.0.0"
-    val binaryVersion = scalaBinaryVersion.value
-    Set(
-      organization.value % s"${moduleName.value}_$binaryVersion" % stableVersion
-    )
+    val rxVersion = """^(\d+)\.(\d+)\.(\d+)(?:-([0-9A-Za-z-]+))?$""".r
+    val previousVersion = version.value match {
+      case rxVersion(major, minor, patch, suffix) if suffix != null =>
+        Some(s"$major.$minor.$patch")
+      case rxVersion(major, "0", "0", null) =>
+        None
+      case rxVersion(major, minor, "0", null) =>
+        Some(s"$major.${minor.toInt - 1}.0")
+      case rxVersion(major, minor, patch, null) =>
+        Some(s"$major.$minor.0")
+      case _ =>
+        sys.error(s"Invalid version number: ${version.value}")
+    }
+    val previousArtifact = {
+      // TODO: Figure out whether there is a more satisfying solution.
+      // Here's what I'd like to do, but I can't because of deprecations:
+      //   val isJVM = crossPlatform.value == JVMPlatform
+      // Here's my second best guess, but it doesn't work due to some reason:
+      //   val isJVM = platformDepsCrossVersion.value == CrossVersion.binary
+      val isJVM = {
+        val isJS = platformDepsCrossVersion.value == ScalaJSCrossVersion.binary
+        val isNative = platformDepsCrossVersion.value == ScalaNativeCrossVersion.binary
+        !isJS && !isNative
+      }
```

Comment:
@sjrd Is there a better way to do this?

Reply:
> Here's my second best guess, but it doesn't work due to some reason

That's because `CrossVersion.binary` is a `def` which returns a new instance of `CrossVersion` every time, and in sbt 0.13.x, `==` is useless on `CrossVersion`s (it's equivalent to `eq`).

There is no elegant way to test the platform of a project in the most general case. The closest you can get is testing for the presence (or absence) of `scalajs-library` or `nativelib` in the `libraryDependencies` of the project.

Alternatively, in your own build, you could define your own very simple `lazy val platform = settingKey[String]` and configure it to your liking.

You could also test whether `thisProject.value.id` (or whatever it's called) ends with `"JVM"`, `"JS"` or `Native`.

And I'm sure you can invent other creative ways to do it.

Reply:
Thanks for the swift response!